lib/game.rb

```diff
@@ -54,8 +55,59 @@ def select_difficulty_level
     ###
   end
 
-  def select_ship_attributes
-    default_attributes = [['Cruiser',3],['Submarine',2]]
+  def select_ship_attributes(board_area, rows, columns)
+
+    @message.would_you_like_to_customize_ships?
+    valid_selections = [1, 2, 3, 4]
+    ship_selection = gets.chomp.to_i
+
+    until valid_selections.include?(ship_selection)
+      @message.invalid_customize_ship_selection
+      ship_selection = gets.chomp.to_i
+    end
+
+    # Ship lengths must be < 1/3 of the board area to play
+    case ship_selection
+    when 1
+      if (board_area / 3) < 5
+        @message.board_is_too_small_for_ship_selection
+        return select_ship_attributes(board_area)
```

Comment:
Will this not trigger a "wrong number of arguments" error?

Should these all be `select_ship_attributes(boar_area, rows, columns)`?

I'm assuming you functionally want to re-start the ship selection process in these cases

Reply:
Yes, fixed.  I originally I thought it only need board_area and then later had to add the rows/columns.